src/System.IO.Ports/src/System/IO/Ports/SerialPort.cs

```diff
@@ -596,6 +598,10 @@ public void Open()
             if (IsOpen)
                 throw new InvalidOperationException(SR.Port_already_open);
 
+            // Check its a valid Serial Port name
+            if (!_portName.Contains("COM") && !_portName.Contains("dev/tty") && !_portName.Contains(GuidDevInterfaceComPort))
```

Comment:
I'm not sure this change is appropriate here - @wfurt, any opinions? I'd vote to remove this check in general and let the native API throw

Reply:
I'd remove the checks- it's what we're doing as a general policy now with `System.IO`. Run the port name through `Path.GetFullPath()` as I've suggested elsewhere. That will throw only for embedded nulls, which you'd want anyway. It will also put the `\\.\` on `COMx` so that addition will need to be removed.

Reply:
Do we also need to worry about culture and case here.
Pretty sure in Windows all Serial ports get names "COMx" in upper case, but not sure about Linux device names, assumed they are all lower case..
I agree with @krwq and just let the Native API throw, as this check might actually stop a valid PortName from being able to be used. Which is worse!

Reply:
I think architecture-specific details like COM vs dev/tty should be separated in their own separate implementations.
Since `SerialPort` is a partial class, I suggest you wrap this condition in a new method defined both in `SerialPort.Windows.cs` and also in a new file `SerialPort.Unix.cs`, then call it here.